dracut.sh

```diff
@@ -1151,6 +1159,15 @@ drivers_dir="${drivers_dir%"${drivers_dir##*[!/]}"}"
 [[ $tmpdir ]] || tmpdir="${dracutsysrootdir-}"/var/tmp
 [[ $INITRD_COMPRESS ]] && compress=$INITRD_COMPRESS
 [[ $compress_l ]] && compress=$compress_l
+[[ $compress_level_l ]] && {
+    compress_level_bzip2=$compress_level_l
+    compress_level_gzip=$compress_level_l
+    compress_level_lz4=$compress_level_l
+    compress_level_lzma=$compress_level_l
+    compress_level_lzop=$compress_level_l
+    compress_level_xz=$compress_level_l
+    compress_level_zstd=$compress_level_l
```

Comment:
Why do we need separate variables for each?

Reply:
The reason is that the different compression levels have different compression level ranges. Imagine setting the compression level to 19, because you use zstd. Then you call `dracut --xz` and the compression level 19 does not exist for xz.

There might be other use cases like: for lz4 I want to use level 1 (to get the fastest compression) but for xz I want the highest compression.

Reply:
> The reason is that the different compression levels have different compression level ranges. Imagine setting the compression level to 19, because you use zstd. Then you call `dracut --xz` and the compression level 19 does not exist for xz.

Oh I see, when you set it in config, and you override it on commandline, you basically want it to reset/revert to some default (or predefined value). That's why there's discrepancy ... you'd need to always override it on commandline as well, off course, as with any config value `dracut --xz --compress-level ...`.

In any case, if switching a compression method on a command line is really that common (while having a compression level defined in a config), let's introduce additonal 7 variables :).